pilot/pkg/security/authn/utils/utils.go

```diff
@@ -30,15 +30,17 @@ import (
 )
 
 // BuildInboundFilterChain returns the filter chain(s) corresponding to the mTLS mode.
-func BuildInboundFilterChain(mTLSMode model.MutualTLSMode, sdsUdsPath string, node *model.Proxy) []networking.FilterChain {
+func BuildInboundFilterChain(mTLSMode model.MutualTLSMode, sdsUdsPath string, node *model.Proxy,
+	listenerProtocol networking.ListenerProtocol) []networking.FilterChain {
 	if mTLSMode == model.MTLSDisable || mTLSMode == model.MTLSUnknown {
 		return nil
 	}
 
 	meta := node.Metadata
 	var alpnIstioMatch *ldsv2.FilterChainMatch
 	var tls *auth.DownstreamTlsContext
-	if util.IsTCPMetadataExchangeEnabled(node) {
+	if util.IsTCPMetadataExchangeEnabled(node) &&
+		(listenerProtocol == networking.ListenerProtocolTCP || listenerProtocol == networking.ListenerProtocolAuto) {
```

Comment:
Could we inline the listenerProtocol check inside the util.IsTCPMetadataExchangeEnabled()?

Reply:
IsTCPMetadataExchangeEnabled is used for constructing upstreamTlsContext too which is a cluster property and hence can't be combined with listener protocol afaik..